src/aarch64/gic.c

```diff
@@ -157,26 +158,70 @@ boolean gic_int_is_pending(int irq)
     return pending;
 }
 
+static void gicr_get_base(void)
+{
+    cpuinfo ci = current_cpu();
+    u64 mpid = mpid_from_cpuid(ci->id);
```

Comment:
Could this not read the mpid directly from the psr?

Reply:
Yes, I changed it to use read_mpid() instead of mpid_from_cpuid().